test/integration/krane_deploy_test.rb

```diff
@@ -1687,6 +1687,15 @@ def test_deploy_globals_without_allow_globals_fails
     ], in_order: true)
   end
 
+  def test_deploy_resource_using_generate_name_succeeds
+    assert_deploy_success(deploy_fixtures("generateName"))
+    pod_name_matcher = /pod-using-generate-name-[a-z0-9]{5}/
+    pods = kubeclient.get_pods(namespace: @namespace)
+    assert_equal(1, pods.count)
+    assert_match(pod_name_matcher, pods.first.dig('metadata', 'name')) # the name contains a generated ID
+    assert_logs_match_all([%r{Pod/pod-using-generate-name-[a-z0-9]{5}\s+Succeeded}])
```

Comment:
Are there a few more times where we log the name earlier in the logs? Let's assert on more of them, since we're changing the value of the name field midway into the deploy.